sympy/functions/special/delta_functions.py

```diff
@@ -218,12 +234,25 @@ def eval(cls, arg):
             return S.Zero
         elif arg.is_positive:
             return S.One
+        elif arg.is_zero and H0 != S.NaN:
+            return H0
```

Comment:
Okay.  This doesn't work well, on first pass.  Leaving `H0 == None` leads to bugs.  `sympy` implicitly assumes function arguments are either built-in number types of sympy objects.   On the other hand, my readings advise against using mutable objects as default keyword values because they introduce subtle hidden-state bugs.